### PR TITLE
Aml tokens

### PR DESCRIPTION
**Pull request by ashleymichal**
Refactors according to discussion in PR #6, which was prematurely merged by myself.

### CODE REVIEW

File: lib/token_generator.rb [hunk 1]
```diff
@@ -1,17 +1,21 @@
 class TokenGenerator
-	@@colors = [:red, :blue, :yellow, :green]
-	@@numbers = (1..12)
+	COLORS = [:red, :blue, :yellow, :green]
+	NUMBERS = (1..12)
+	TOKEN = Struct.new :color, :number do
```
Comment (dmin) [inline lib/token_generator.rb]: Looking good. The only thing I might suggest is to rename ```TOKEN``` to ```Token``` - its a pretty standard convention to name a Struct object like you would a class (first letter uppercase).

**Comment by ashleymichal** [comment]
Good call. The last thought I have here is this: should TokenGenerator return an Array of Token structs (like it does now), or a TokenBag object composed of Token structs? I'm content to leave this as is until we decide that there is behavior that belongs in a TokenBag object.  I'm kind of interested in building the logic for validating a Run and seeing what kinds of requirements that creates.

**Comment by dmin** [comment]
Yeah, I'd agree that just returning an array of Token objects is a good option for now. Yeah - if you want to work on run validation, go for it!

**Comment by dmin** [comment]
@ashleymichal Just noticed that you merged this into the "aml-tokens" branch, not master. Is that what you wanted?

**Pull request merged**